wfl/Controllers/RestaurantsController.cs

```diff
@@ -1,42 +1,31 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Web;
-using System.Web.Http.Description;
-using System.Web.Http;
-using System.Web.Http.Results;
-using wfl.Models;
+﻿using System.Web.Http;
+using System.Threading.Tasks;
+using Symphono.Wfl.Models;
+using Symphono.Wfl.Database;
 
 
-namespace wfl.Controllers
+namespace Symphono.Wfl.Controllers
 {
     [RoutePrefix("api/Restaurant")]
     public class RestaurantsController : ApiController
     {
         [Route("")]
         [HttpPost]
-        [ResponseType(typeof(Restaurant))]
-        public IHttpActionResult CreateRestaurant([FromUri] string name)
+        public async Task<IHttpActionResult> CreateRestaurantAsync([FromBody] RestaurantDto restaurant)
         {
-            if (name == null || name.Equals(""))
+            if (restaurant == null || string.IsNullOrEmpty(restaurant.Name))
```

Comment:
FYI, you can use null coalescence here to do 
```
string.IsNullOrEmpty(restaraunt?.Name)
```

Notice the `?` which means if the value is not null, use the next expression, otherwise return null.